### PR TITLE
XS-2883 - Extract if check for a dynamic message

### PR DESCRIPTION
**Pull request by brianmyers-wf**
#### Changes:
- Pulled nested if logic around message codes out of `_build_id_messages` function and into new helper function called `_get_message_codes`.
- Added a couple unit tests to ensure new function is getting tested.

Please review: @hefischer  @andrewperkins-wf

### CODE REVIEW

**Comment by jasonleinbach-wf** [comment]
+1

File: utilities/generate_messages_catalog.py [hunk 1]
```diff
@@ -291,6 +281,30 @@ def _build_id_messages(python_module):
...
 
 
+def _get_message_codes(msg_code_arg):
+    """
+    Get the correct message codes based on instance type of msgCodeArg
+
+    :param msg_code_arg: the current arg to pull msgCodes from
+    :type msg_code_arg: :class:`~ast.AST`
+    :return: the correct message code to use
+    :rtype: str
```
Comment (chrislococo-wf) [inline utilities/generate_messages_catalog.py]: CR: It looks like msg_codes is a set of strings, so I think this should be `:rtype: set (str)`?

File: utilities/generate_messages_catalog.py [hunk 2]
```diff
@@ -291,6 +281,30 @@ def _build_id_messages(python_module):
...
+        msg_codes = (msg_code_arg.s,)
+    else:
+        if any(isinstance(element, (ast.Call, ast.Name))
+               for element in ast.walk(msg_code_arg)):
+            msg_codes = ('(dynamic)',)
+        else:
+            msg_codes = [
+                element.s for element in ast.walk(msg_code_arg)
+                if isinstance(element, ast.Str)
+            ]
```
Comment (chrislococo-wf) [inline utilities/generate_messages_catalog.py]: For consistency, this should be cast to a set.

File: utilities/generate_messages_catalog.py (hunk 1 above)
Comment (brianmyers-wf) [inline utilities/generate_messages_catalog.py]: you're right, it's not just a string, but is it a set? looks more like a tuple...

File: utilities/generate_messages_catalog.py (hunk 2 above)
Comment (brianmyers-wf) [inline utilities/generate_messages_catalog.py]: and if it is a tuple, isn't a list and tuple basically the same?

File: utilities/generate_messages_catalog.py (hunk 1 above)
Comment (chrislococo-wf) [inline utilities/generate_messages_catalog.py]: Yep, tuple, not set, sorry about that

File: utilities/generate_messages_catalog.py (hunk 2 above)
Comment (chrislococo-wf) [inline utilities/generate_messages_catalog.py]: No, tuples are static and cannot be modified.

**Comment by jasonleinbach-wf** [comment]
+1 refresh

**Comment by chrislococo-wf** [comment]
+1

**Comment by jasonleinbach-wf** [comment]
+1

**Comment by chrislococo-wf** [comment]
+1

**Comment by chrislococo-wf** [comment]
#### QA Steps:
- Manual testing: Generation of messages succeeded.
- Flake8: 

```
Not yet. TODO.
```
- Nosetest result: 

```
(builder)  chrislococo [arelle_builder] (XS-2883) $ nosetests test/
...........
----------------------------------------------------------------------
Ran 11 tests in 0.009s

OK
```

#### Result: QA +10

**Pull request merged**